src/lgdo/types/struct.py

```diff
@@ -89,20 +81,68 @@ def form_datatype(self) -> str:
     def update_datatype(self) -> None:
         self.attrs["datatype"] = self.form_datatype()
 
-    def add_field(self, name: str | int, obj: LGDO) -> None:
+    def add_field(self, name: str | int, obj: LGDO | Mapping) -> None:
         """Add a field to the table."""
```

Comment:
this docstring is quite bad, could you improve it? see also the Table.add_field() docstring.